adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,225 @@
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is coverage of the first base pair in that bin.
+   *
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def coverage(bpPerBin: Int = 1): CoverageRDD = {
+
+    val flattened = this.flatten
+
+    if (bpPerBin == 1) {
+      flattened // no binning, return raw results
+    } else {
+      // subtract region.start to shift mod to start of ReferenceRegion
+      val newRDD = flattened.rdd.filter(r => r.start % bpPerBin == 0)
+      flattened.replaceRdd(newRDD)
+    }
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is the mean coverage over all base pairs in that bin.
+   *
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def aggregatedCoverage(bpPerBin: Int = 1): CoverageRDD = {
+
+    val flattened = this.flatten
+
+    def reduceFn(a: (Double, Int), b: (Double, Int)): (Double, Int) = {
+      (a._1 + b._1, a._2 + b._2)
+    }
+
+    if (bpPerBin == 1) {
+      flattened // no binning, return raw results
+    } else {
+      val newRDD = flattened.rdd
+        .keyBy(r => {
+          // key coverage by binning start site mod bpPerbin
+          // subtract region.start to shift mod to start of ReferenceRegion
+          val start = r.start - (r.start % bpPerBin)
+          ReferenceRegion(r.contigName, start, start + bpPerBin)
+        }).mapValues(r => (r.count, 1))
+        .reduceByKey(reduceFn)
+        .map(r => {
+          // compute average coverage in bin
+          Coverage(r._1.referenceName, r._1.start, r._1.end, r._2._1 / r._2._2)
+        })
+      flattened.replaceRdd(newRDD)
+    }
+  }
+
+  /**
+   * @param elem The Coverage to get an underlying region for.
+   * @return Since coverage maps directly to a single genomic region, this
+   *   method will always return a Seq of exactly one ReferenceRegion.
+   */
+  protected def getReferenceRegions(elem: Coverage): Seq[ReferenceRegion] = {
+    Seq(ReferenceRegion(elem.contigName, elem.start, elem.end))
+  }
+
+  /**
+   * @param newRdd The RDD to replace the underlying RDD with.
+   * @return Returns a new CoverageRDD with the underlying RDD replaced.
+   */
+  protected def replaceRdd(newRdd: RDD[Coverage]): CoverageRDD = {
+    copy(rdd = newRdd)
+  }
+
+  /**
+   * Gets flattened RDD of coverage, with coverage mapped to a reference Region at each base pair
+   * @return CoverageRDD of flattened Coverage records
+   */
+  def flatten: CoverageRDD = {
+    this.replaceRdd(flatMapCoverage(rdd))
```

Comment:
No `this.`, same comment above RE: transform --> this becomes `transform(flatMapCoverage)`